src/electionguard/scheduler.py

```diff
@@ -0,0 +1,103 @@
+from __future__ import annotations
+from contextlib import AbstractContextManager
+from multiprocessing import Pool as ProcessPool
+from multiprocessing.dummy import Pool as ThreadPool
+from multiprocessing.pool import Pool
+from psutil import cpu_count
+import traceback
+from typing import Any, Callable, Iterable, List
+
+from .logs import log_warning
+from .singleton import Singleton
+from .utils import T
+
+
+class Scheduler(Singleton, AbstractContextManager):
```

Comment:
this implementation is generic enough that it should be straightforward to override this pattern and implement multiprocessing in some other way - such as via a cluster.  in the future we could turn this into a protocol and allow for alternative implementations but that is beyond the scope of this work